app/repositories/application/GeneralApplicationRepository.scala

```diff
@@ -364,15 +365,30 @@ class GeneralApplicationMongoRepository(timeZoneService: TimeZoneService,
       BSONDocument("applicationStatus" -> BSONDocument("$ne" -> WITHDRAWN))
     )))
 
-  override def findFailedTestForNotification(failedTestType: FailedTestType): Future[Option[NotificationFailedTest]] = {
-    val query = BSONDocument("$and" -> BSONArray(
-      BSONDocument("applicationStatus" -> failedTestType.appStatus),
-      BSONDocument(s"progress-status.${failedTestType.notificationProgress}" -> BSONDocument("$ne" -> true)),
-      BSONDocument(s"progress-status.${failedTestType.receiveStatus}" -> true)
-    ))
+  override def findTestForNotification(notificationType: NotificationTestType): Future[Option[TestResultNotification]] = {
+    val query = Try{ notificationType match {
```

Comment:
I'm happy to l.eave it in if we think we're going to add more notification types then